File: java/src/main/java/com/dddeurope/niuz/finance/Contract.java
package com.dddeurope.niuz.finance;

public class Contract {
    private final String authorId;
    private final String contractType;
    private final Integer rate;
    private final String bankAccount;
    private final String authorName;

    public Contract(String authorId, String contractType, Integer rate, String bankAccount, String authorName) {

        this.authorId = authorId;
        this.contractType = contractType;
        this.rate = rate;
        this.bankAccount = bankAccount;
        this.authorName = authorName;
    }

    public boolean IsOwnedBy(String authorId) {
        return this.authorId.equals(authorId);
    }

    public Payment GeneratePayment(String description) {
        return new Payment(rate, bankAccount, authorName, description);
    }

    public boolean paysBySubmission() {
        return contractType.equals("pay-by-submission");
    }

    public boolean paysByPublication() {
        return contractType.equals("pay-by-publication");
    }
}

File: dotnet/niuz.application/finance/Contract.cs
namespace niuz.application.finance
{
    public class Contract
    {
        private readonly string authorId;
        private readonly int rate;
        private readonly string bankAccount;
        private readonly string authorName;

        public Contract(string authorId, string contractType, int rate, string bankAccount, string authorName)
        {
            this.authorId = authorId;
            this.PaysBySubmission = contractType == "pay-by-submission";
            this.PaysByPublication = contractType == "pay-by-publication";

            this.rate = rate;
            this.bankAccount = bankAccount;
            this.authorName = authorName;
        }

        public bool PaysByPublication { get; }
        public bool PaysBySubmission { get; }

        public bool IsOwnedBy(string authorId)
        {
            return this.authorId == authorId;
        }

        public Payment GeneratePayment(string description)
        {
            return new Payment(rate, bankAccount, authorName, description);
        }
    }
}